src/infrastructure/Models/Routing/RouteCollectionBuilder.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Infrastructure\Models\Routing;
+
+use Infrastructure\Exceptions\InfrastructureException;
+use Symfony\Component\Routing\RouteCollection;
+
+class RouteCollectionBuilder
+{
+    /**
+     * @var RouteCollection
+     */
+    private $routeCollection;
+
+    /**
+     * RouteCollectionBuilder constructor.
+     */
+    public function __construct()
+    {
+        $this->routeCollection = new RouteCollection();
+    }
+
+    /**
+     * @param string $path
+     * @param string $presentationService
+     * @return RouteCollectionBuilder
+     * @throws InfrastructureException
+     */
+    public function addCRUD(string $path, string $presentationService) : RouteCollectionBuilder
```

Comment:
1) Your controller service doesn't  have any contract that you can rely on methods you provide. Ex. Update, delete
2) Such implementation forces you to develop this methods from the box. What if you don't need that, that you should have ability to specify methods and routes you need.  Ex. url only to load.
3) Specific resource, ex. send email. Which can be treated as create /email or specific sub resource. And this can be specific service function, ex. sendEmail().
4) as framework user, looks quite unfriendly to put restrictions what urls can exists on resource in my app.

Reply:
The purpose was rid of Symfony ugly syntax.

It's just syntax sugar

You can use:

````
$routesCollectionBuilder = new RouteCollectionBuilder();

$routesCollectionBuilder->addCRUD('/restaurants', Restaurant::class);

$routeCollection = $routesCollectionBuilder->build();

$routeCollection->add('whateverYouWant',
    (new RouteBuilder())
        ->setPath('you/url')
        ->setService('YouService')
        ->setServiceMethod('youMethod')
        ->build()
);

return $routeCollection;

````

Reply:
I've added small description to the readme